### PR TITLE
Member edit page

### PR DESCRIPTION
**Pull request by jboning**
Things not done here:
 * Make the forms prettier. (Getting a one-line form by using the table markup without a surrounding `<table>` element is shameful and ugly.)
 * Phone/email "pagable" settings should be gated by confirmation that the page will be received (and pagable phones/emails shouldn't be editable through this UI).
 * Tests.

### CODE REVIEW

File: main/views/member_views.py [hunk 1]
```diff
@@ -39,6 +40,116 @@ class MemberDetailView(LoginRequiredMixin, generic.DetailView):
     template_name = 'member_detail.html'
 
 
+class MemberEditView(LoginRequiredMixin, generic.base.TemplateView):
+    template_name = 'member_form.html'
+
+    MemberForm = modelform_factory(Member,
+            fields=['first_name', 'last_name', 'ham', 'v9', 'dl'])
+    PhonesForm = inlineformset_factory(Member, Phone,
+            fields=['type', 'number', 'pagable', 'sms_email', 'position'],
```
Comment (kduncklee) [inline main/views/member_views.py]: We can get rid of sms_email. We let the sms service handle it.

File: main/views/member_views.py [hunk 2]
```diff
@@ -39,6 +40,116 @@ class MemberDetailView(LoginRequiredMixin, generic.DetailView):
...
+                #print("ordered forms:")
+                #for i, individual_form in enumerate(f.ordered_forms):
+                #    obj = individual_form.save(commit=False)
+                #    obj.position = i
+                #    print(obj, individual_form.has_changed())
+                #    #obj.save()
+                #print("deleted:")
+                #for obj in f.deleted_objects:
+                #    print(obj)
+                ##    obj.delete()
```
Comment (kduncklee) [inline main/views/member_views.py]: Can this commented block be deleted?

File: main/static/js/dynamic_formset.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Based on https://gist.github.com/taranjeet/71b7826b60f42e5d239cf3b3abbf292f
+
+/*
+function updateElementIndex(el, prefix, ndx) {
```
Comment (kduncklee) [inline main/static/js/dynamic_formset.js]: Remove this if it's not needed?

File: main/views/member_views.py (hunk 1 above)
Comment (jboning) [reply main/views/member_views.py]: OK, great. I sort of suspected this wouldn't be needed.

File: main/views/member_views.py (hunk 2 above)
Comment (jboning) [reply main/views/member_views.py]: Yep, whoops.

File: main/static/js/dynamic_formset.js (hunk 3 above)
Comment (jboning) [reply main/static/js/dynamic_formset.js]: I thought I pushed another commit that cleaned this up, let me see what happened there.

File: main/views/member_views.py [hunk 4]
```diff
@@ -39,6 +40,116 @@ class MemberDetailView(LoginRequiredMixin, generic.DetailView):
...
+            # errors. Setting autocomplete off makes the browser leave
+            # TOTAL_FORMS alone.
+            for f in forms.values():
+                if hasattr(f, 'management_form'):
+                    f.management_form.fields['TOTAL_FORMS'].widget.attrs['autocomplete'] = 'off'
+            self.forms = forms
+        return self.forms
+
+    def post(self, *args, **kwargs):
+        if self.kwargs['pk'] != self.request.user.id:
```
[the remark this answers is not in the document]
Comment (jboning) [reply main/views/member_views.py]: Ah right, I'll add a TODO here again. Probably time to address this soon.

**Pull request merged**